w11h03/test/pgdp/networking/NetworkTest.java

```diff
@@ -0,0 +1,319 @@
+    //
+
+    @BeforeAll
+    public static void init() throws Exception {
+        var config = Files
+                .lines(Path.of("test/config.env"))
+                .filter(i -> i.trim().length() != 0)
+                .filter(i -> !i.startsWith("#"))
+                .collect(Collectors.toMap(i -> i.split("=")[0].trim(), i -> i.split("=")[1].trim()));
+
+        kennung = config.get("kennung");
+        username = config.get("username");
+        password = config.get("password");
+
+        if (config.get("kennung").equals("null")) {
+            System.err.println("""
+                    [STEP 1] To setup the Tests, please provide your
+                    TUM kennung in the `config.env` file and run the
+                    tests again. This execution all tests will fail.
+                    """);
+            return;
+        }
+
+        if (config.get("username").equals("null")) {
+            System.err.println("""
+                    [STEP 2] Please provide a username to your
+                    config file.
+                    """);
+            return;
+        }
+
+        if (config.get("password").equals("null")) {
+            assertTrue(dataHandler.register(username, kennung));
+
+            System.err.println("""
+                    [STEP 3] An email was sent to your mailbox
+                    associated with the TUM kennung. Please open
+                    it and provide the config file with the pass-
+                    word you have received.
+                    """);
+            return;
+        }
+
+        executeTests = true;
+        System.out.println("RUNNING TESTS...");
+    }
+
+    @BeforeEach
+    public void configError() {
+        if (!executeTests)
+            fail("Please follow the instructions to set up your test files.");
+    }
+
+    //
+    // BEHAVIOUR TEST
+    //
+
+    //
+    // FLAGS
+    //
+    // [A] - Test that should <A>lawys pass and not be affected by code.
+    // [P] - Test in <P>ublic environment.
+    // [T] - <T>emplate can be broken, ensure this.
+    //
+
+    //
+    // <region structural tests>
+    //
+
+    @Test
+    @Order(1)
+    @DisplayName("[T] should be able to login with credentials")
+    public void loginTest() throws Exception {
+        assertTrue(dataHandler.login(username, password));
+
+        id = (int) Lib.getField(dataHandler, "id").get(dataHandler);
+        System.out.println("ID: " + id);
+
+        assertNotNull(id);
+        assertTrue(id > 0);
+
+        assertEquals(username, (String) Lib.getField(dataHandler, "username").get(dataHandler));
+        assertEquals(password, (String) Lib.getField(dataHandler, "password").get(dataHandler));
+    }
+
+    @Test
+    @Order(2)
+    @DisplayName("should request token")
+    public void requestTokenTest() {
+        var token = dataHandler.requestToken();
+        assertNotNull(token);
+        System.out.println("TOKEN: " + token);
+    }
```

Comment:
Isn't this test always false because `requestToken` returns false because `username` and `password` are null?

Reply:
See Artemis Task for login

> In der login … übergebenen Nutzernamen und Passwort anfragen. Falls diese Anfrage erfolgreich ist, werden … die entsprechenden Klassenattribute gesetzt …

username and password should not be null after Order(1) Test

Reply:
Can we add a comment that you need to run all tests at once?

Reply:
https://github.com/MaximilianAnzinger/pgdp2223-tests/pull/277/commits/3e4b27e0e86210626633792cd29838c16c5491a9